### ISSUE TITLE
Parameter $refs not available

### ISSUE THREAD

**Issue by mikefeher-waitr** [apigee-127/swagger-test-templates#104]
Currently does not support derefrencing of the parameter values. There is a [check failing](https://github.com/apigee-127/swagger-test-templates/blob/master/index.js#L136) on $refs for parameters on the path level.  Not sure how to approach this without dereferencing the doc and copying the output of the path level parameter as a work around. Just as a side note, I switched over to the [BigStickCarpet](https://github.com/BigstickCarpet/json-schema-ref-parser) derefrencer library since a lot of the libraries are hit and miss with keeping up-to spec with JSON Schema Draft 4/5 | OpenAPI/Swagger 2.0.

**Comment by melindafekete** [comment]
I had to add these to make it work with current swagger specs

```js
ZSchema.registerFormat("int32", function (str) {
  return !isNaN(str);
});
ZSchema.registerFormat("double", function (str) {
  return !isNaN(str);
});

// this register the specs on the validator, I couldn't get it to work with just '' for the reference 
// which is annoying because you need to modify the "$ref" to add the REFERENCE you chose. But maybe there is a way.

validator.setRemoteReference('REFERENCE', require('path/to/swaggerspec.json'));
```

and then using 
```
 "$ref": "REFERENCE#/definitions/User"
```

worked for me

**Comment by Remco75** [comment]
@melindafekete , thnx. Seems that this module is not ready for the new specs. 
More formats are missing, (I encountered int64, but there could be more)

I will try to fix this in a PR, but it will take time. Help appreciated ;-)

**Comment by Remco75** [comment]
btw there seem to be 2 problems:
- I think in the generated test the schema variable is meant to be populated with the actual schema-json (instead of the ref)
- the missing types need to be declared in z-schema automaticly

**Comment by Remco75** [comment]
hmm, some update, there is code in master that parses the $refs, but it is not includes in the npm-release :-(

Please release the latest code!

**Comment by Remco75** [comment]
#110

**Comment by noahdietz** [comment]
Fixed by #110. Closing.
